### ISSUE TITLE
Add more to the output of Assert.Single

### ISSUE THREAD

**Issue by g0t4** [xunit/xunit#1351]
When I call `Assert.Single(list, "expectedValue");` I get the following on a failure:
```
Message: The collection contained 0 matching element(s) instead of 1.
```
It would be helpful to know which `expectedValue` wasn't found, especially if a test contains multiple `Assert.Single` assertions. 

Perhaps:

```
Message: Expected 1 of "expectedValue" but collection contains X
```

Or, based on the number of matches, customize the message:

```
Collection is missing "expectedValue"
Collection should only have 1 of "expectedValue" but found 2
```

Perhaps even print out the collection like `Assert.Contains` does, if the collection isn't too big?

Also, would it be possible to use an expression for the predicate and walk the expression to print it out too? That might be a bit much.

**Comment by davidwengier** [comment]
I have opened a couple of PRs for this, though with slightly different messages than suggested. I looked at reusing the Contains style messages but succinctly communicating the notion of "one and only one item wasn't found" in that format was beyond me.

**Comment by bradwilson** [comment]
Fixed with #1837 and xunit/assert.xunit#38
